rstudio/rstudio
# Minimize panes without tabs and show panes with newly added tabs

### Intent

Fixes issue addressed in #8460 

1. When a user updates their pane layout through Global Options or `rstudio-prefs.json`, hide newly empty tab panes or show panes that were previously empty but now have tabs. 
2. Updated code that restores panes to "default" in several cases. Previously we always restored a pane's `WindowState` to `NORMAL`, this update sets the state to `MINIMIZE` if the pane is an empty tab pane. 

### Approach

1. Replace code that iterated over all panes and set their state to normal with a function that determines if the state should be `NORMAL` or `MINIMIZED`. 
2. Update `panes` user preference `ValueChangeHandler` to check if a tab set pane has newly become emptied or possible and if so, update the state accordingly. 

### Automated Tests

https://github.com/rstudio/rstudio-ide-automation/issues/150

### QA Notes

1. See #8460 
2. Configure your layout so one tab pane is completely empty, then zoom into any non-empty pane. Zoom the pane a second time to return to your original configuration and the hidden pane should remain hidden.
